tests/balloonpanel/balloonpanelview.js

```diff
@@ -0,0 +1,234 @@
+			expect( view.element.tagName ).to.equal( 'DIV' );
+			expect( view.element.classList.contains( 'ck-balloon-panel' ) ).to.true;
+			expect( view.element.classList.contains( 'ck-link-balloon-panel' ) ).to.true;
+		} );
+
+		it( 'should registers "content" region', () => {
+			expect( view.regions.get( 0 ).name ).to.equal( 'content' );
+			expect( view.regions.get( 0 ).element ).to.equal( view.element );
+		} );
+	} );
+
+	describe( 'keyboard support', () => {
+		it( 'should handle `keydown` event only when balloon is open', () => {
+			const escPressSpy = sinon.spy( view, '_closeOnEscPress' );
+
+			view.model.isVisible = true;
+
+			document.dispatchEvent( new Event( 'keydown' ) );
+
+			expect( escPressSpy.calledOnce ).to.true;
+
+			view.model.isVisible = false;
+
+			document.dispatchEvent( new Event( 'keydown' ) );
+
+			// Still called only once.
+			expect( escPressSpy.calledOnce ).to.true;
+		} );
+
+		it( 'should hide balloon on `Esc` press', () => {
+			const hideSpy = sinon.spy( view, 'hide' );
+			const fakeEventInfo = {};
+			const fakeKeydownEvent = { keyCode: 27 };
+
+			view._closeOnEscPress( fakeEventInfo, fakeKeydownEvent );
+
+			view.model.isVisible = false;
+
+			expect( hideSpy.calledOnce ).to.true;
+		} );
+	} );
+
+	describe( 'DOM bindings', () => {
+		describe( 'arrow', () => {
+			it( 'should reacts on view.model#arrow', () => {
+				expect( view.element.classList.contains( 'ck-balloon-panel_arrow_se' ) ).to.true;
+
+				view.model.set( 'arrow', 'sw' );
+
+				expect( view.element.classList.contains( 'ck-balloon-panel_arrow_sw' ) ).to.true;
+			} );
+		} );
+
+		describe( 'isVisible', () => {
+			it( 'should reacts on view.model#isvisible', () => {
+				expect( view.element.classList.contains( 'ck-balloon-panel_visible' ) ).to.false;
+
+				view.model.isVisible = true;
+
+				expect( view.element.classList.contains( 'ck-balloon-panel_visible' ) ).to.true;
+			} );
+		} );
+
+		describe( 'styles', () => {
+			it( 'should reacts on view.model#top', () => {
+				expect( view.element.style.top ).to.equal( '0px' );
+
+				view.model.top = 10;
+
+				expect( view.element.style.top ).to.equal( '10px' );
+			} );
+
+			it( 'should reacts on view.model#left', () => {
+				expect( view.element.style.left ).to.equal( '0px' );
+
+				view.model.left = 10;
+
+				expect( view.element.style.left ).to.equal( '10px' );
+			} );
+
+			it( 'should reacts on view.model#maxWidth', () => {
+				expect( view.element.style.maxWidth ).to.equal( '200px' );
+
+				view.model.maxWidth = 10;
+
+				expect( view.element.style.maxWidth ).to.equal( '10px' );
+			} );
+		} );
+	} );
+
+	describe( 'isVisible', () => {
+		it( 'should return view.model#isVisible value', () => {
+			expect( view.isVisible ).to.false;
+
+			view.model.isVisible = true;
+
+			expect( view.isVisible ).to.true;
+		} );
+	} );
+
+	describe( 'show', () => {
+		it( 'should set view.model#isVisible as true', () => {
+			view.show();
```

Comment:
I'd set `view.model.isVisible` to `false` here first. Otherwise, it's hardly asserting anything.